### PR TITLE
edit project structure, build manager - gradle

### PR DESCRIPTION
**Pull request by sergeyarhutik**
edit project structure, build manager - gradle.

### CODE REVIEW

**Review by RomanukAlex — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -128,4 +128,9 @@ buildNumber.properties
 .mvn/timing.properties
```
Comment (aleksieiev) [inline .gitignore]: looks like we do not need this any more

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        springBootVersion 	= '2.2.0.RELEASE'
+        postgreSqlVersion = '42.2.8'
+        junitVersion 		= '4.12'
+        restAssuredVersion 	= '2.9.0'
+    }
+
+    repositories {
+        mavenCentral()
+        maven { url "https://repo.spring.io/snapshot" }
+        maven { url "https://repo.spring.io/milestone" }
```
Comment (aleksieiev) [inline build.gradle]: what for?

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+subprojects {
+    apply plugin: 'java'
+    apply plugin: 'maven'
+    apply plugin: 'idea'
+    apply plugin: 'jacoco'
+    apply plugin: 'io.spring.dependency-management'
+
+    sourceCompatibility = 1.8
+    targetCompatibility = 1.8
```
Comment (aleksieiev) [inline build.gradle]: Java 11?

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    repositories {
+        mavenCentral()
+        maven { url "https://repo.spring.io/snapshot" }
+        maven { url "https://repo.spring.io/milestone" }
```
Comment (aleksieiev) [inline build.gradle]: what for?

File: build.gradle [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    dependencyManagement {
+        imports {
+            mavenBom("org.springframework.boot:spring-boot-dependencies:2.2.0.RELEASE")
```
Comment (aleksieiev) [inline build.gradle]: version as a constant

File: build.gradle [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        compile "org.springframework.boot:spring-boot-starter"
+        compile "org.springframework.boot:spring-boot-starter-data-jdbc"
+        compile "org.springframework.boot:spring-boot-starter-data-jpa"
+        compile "org.springframework.boot:spring-boot-starter-thymeleaf"
+        compile "org.springframework.boot:spring-boot-starter-web"
+        compile "org.springframework.boot:spring-boot-starter-web-services"
+        compile "com.h2database:h2:"
+        compile "org.postgresql:postgresql:${postgreSqlVersion}"
+        compile "mysql:mysql-connector-java:${mysqlConnectorJavaVersion}"
+        compile "com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.10.0"
```
Comment (aleksieiev) [inline build.gradle]: version as a constant

File: build.gradle [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    dependencies {
+        compile "org.springframework.boot:spring-boot-starter"
+        compile "org.springframework.boot:spring-boot-starter-data-jdbc"
+        compile "org.springframework.boot:spring-boot-starter-data-jpa"
+        compile "org.springframework.boot:spring-boot-starter-thymeleaf"
+        compile "org.springframework.boot:spring-boot-starter-web"
+        compile "org.springframework.boot:spring-boot-starter-web-services"
+        compile "com.h2database:h2:"
+        compile "org.postgresql:postgresql:${postgreSqlVersion}"
+        compile "mysql:mysql-connector-java:${mysqlConnectorJavaVersion}"
```
Comment (aleksieiev) [inline build.gradle]: 1. maybe parent should resolve version?
2. This is a subproject dependency section. Why do you think that all subprojects should have these dependencies?

File: build.gradle [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    dependencies {
+        compile "org.springframework.boot:spring-boot-starter"
+        compile "org.springframework.boot:spring-boot-starter-data-jdbc"
+        compile "org.springframework.boot:spring-boot-starter-data-jpa"
+        compile "org.springframework.boot:spring-boot-starter-thymeleaf"
+        compile "org.springframework.boot:spring-boot-starter-web"
+        compile "org.springframework.boot:spring-boot-starter-web-services"
+        compile "com.h2database:h2:"
```
Comment (aleksieiev) [inline build.gradle]: Ugly. Last `:` isn't required

File: dao-api/build.gradle [hunk 9]
```diff
@@ -0,0 +1,5 @@
+apply plugin: 'java'
+dependencies {
+  compile project(':model')
+  compile "org.springframework:spring-context"
```
Comment (aleksieiev) [inline dao-api/build.gradle]: what for? think it's included in some pf the starters

File: rest-app/build.gradle [hunk 10]
```diff
@@ -0,0 +1,5 @@
+apply plugin: 'java'
+dependencies {
+  compile project(':service')
+    compile group: 'org.springframework.boot', name: 'spring-boot-starter', version:'2.2.0.RELEASE'
```
Comment (aleksieiev) [inline rest-app/build.gradle]: 1. version shouldn't been declared here.
2. this dependency already assigned on a top level for all of subprojects

File: settings.gradle [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+include ':dao-api'
+include ':service'
+include ':service-api'
+include ':rest-app'
+
+project(':model').projectDir = "$rootDir/model" as File
+project(':dao-api').projectDir = "$rootDir/dao-api" as File
+project(':service').projectDir = "$rootDir/service" as File
+project(':service-api').projectDir = "$rootDir/service-api" as File
+project(':rest-app').projectDir = "$rootDir/rest-app" as File
```
Comment (aleksieiev) [inline settings.gradle]: what for?
Redundant.

**Review by aleksieiev — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 3 above)
Comment (sergeyarhutik) [reply build.gradle]: - failed to read class file /home/siarhei_arkhutsik/IdeaProjects/repair-service_gradle/dao-api/build/classes/java/test/com/epam/repair/dao/TestConfig.class
- Unsupported class file major version 55

with java 11.

Comment (aleksieiev) [reply build.gradle]: ... and what?

File: build.gradle (hunk 7 above)
Comment (sergeyarhutik) [reply build.gradle]: should I put this dependencies to dao-api?

Comment (aleksieiev) [reply build.gradle]: `-api`? Why API?

**Review by RomanukAlex — APPROVED** [review]

File: dao-api/build.gradle [hunk 12]
```diff
@@ -0,0 +1,4 @@
+apply plugin: 'java'
```
Comment (aleksieiev) [inline dao-api/build.gradle]: Nahoa?
Parent `build.gradle` already applied this plugin

File: model/build.gradle [hunk 13]
```diff
@@ -0,0 +1,6 @@
+
+apply plugin: 'java'
```
Comment (aleksieiev) [inline model/build.gradle]: the same here
redundant

File: rest-app/build.gradle [hunk 14]
```diff
@@ -0,0 +1,9 @@
+apply plugin: 'java'
+dependencies {
+  def postgreSqlVersion = '42.2.8'
+  def mysqlConnectorJavaVersion = '8.0.18'
```
Comment (aleksieiev) [inline rest-app/build.gradle]: shouldn't been here

**Review by aleksieiev — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 15]
```diff
@@ -0,0 +1,70 @@
+buildscript {
+    ext {
+        projectGroupName 	= 'com.epam.repair'
+        projectVersion 		= '1.0-SNAPSHOT'
+        springBootVersion 	= '2.2.0.RELEASE'
+        junitVersion 		= '4.12'
```
Comment (aleksieiev) [inline build.gradle]: junit 5

**Review by RomanukAlex — APPROVED** [review]

File: build.gradle [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+
+    dependencies {
+        compile "org.springframework.boot:spring-boot-starter"
+        compile "org.springframework.boot:spring-boot-starter-data-jdbc"
+        compile "org.springframework.boot:spring-boot-starter-data-jpa"
+        compile "org.springframework.boot:spring-boot-starter-thymeleaf"
+        compile "org.springframework.boot:spring-boot-starter-web"
+        compile "org.springframework.boot:spring-boot-starter-web-services"
+        compile "com.h2database:h2"
+        compile "com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:${fasterxmlJacksonVersion}"
```
Comment (aleksieiev) [inline build.gradle]: Why other dependencies doesn't require version, but for this it's required?

File: dao-api/build.gradle [hunk 17]
```diff
@@ -0,0 +1,5 @@
+dependencies {
+    compile project(':model')
+    compile "org.postgresql:postgresql:${postgreSqlVersion}"
+    compile "mysql:mysql-connector-java:${mysqlConnectorJavaVersion}"
```
Comment (aleksieiev) [inline dao-api/build.gradle]: version is redundant

**Review by aleksieiev — CHANGES_REQUESTED** [review]

**Review by aleksieiev — APPROVED** [review]

**Pull request merged**